cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -897,18 +894,45 @@ func (m *recoverModeStateMachine) unlockSaveRunKey() (stateFunc, error) {
 	}
 	if unlockErr != nil {
 		// failed to unlock with run key, try fallback key
-		return m.unlockSaveFallbackKey, nil
+		return m.unlockEncryptedSaveFallbackKey, nil
 	}
 
 	// unlocked it properly, go mount it
 	return m.mountSave, nil
 }
 
-func (m *recoverModeStateMachine) unlockSaveFallbackKey() (stateFunc, error) {
-	// remember what we assumed about encryption before looking at
-	// save
-	assumeEncrypted := m.isEncryptedDev
+func (m *recoverModeStateMachine) unlockMaybeEncryptedSaveFallbackKey() (stateFunc, error) {
+	// first check whether there is an encrypted save
+	_, findErr := m.disk.FindMatchingPartitionUUIDWithFsLabel("ubuntu-save-enc")
```

Comment:
do we use the name "-enc" anywhere else here?  it seems that name should be produced with the help of our secboot somehow